instrumentation/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -308,7 +315,32 @@ private SdkTracerProvider buildTracerProvider(SessionId sessionId, Application a
     private SpanExporter buildSpanExporter() {
         // TODO: Default to otlp...but how can we make endpoint and auth mandatory?
         SpanExporter defaultExporter = LoggingSpanExporter.create();
-        return spanExporterCustomizer.apply(defaultExporter);
+        SpanExporter spanExporter = defaultExporter;
+        DiskBufferingConfiguration diskBufferingConfiguration =
+                config.getDiskBufferingConfiguration();
+        if (diskBufferingConfiguration.isEnabled()) {
+            try {
+                spanExporter = createDiskExporter(defaultExporter, diskBufferingConfiguration);
+            } catch (IOException e) {
+                Log.w(RumConstants.OTEL_RUM_LOG_TAG, "Could not create span disk exporter.", e);
+            }
+        }
+        return spanExporterCustomizer.apply(spanExporter);
+    }
+
+    private static SpanExporter createDiskExporter(
+            SpanExporter defaultExporter, DiskBufferingConfiguration diskBufferingConfiguration)
+            throws IOException {
```

Comment:
Is it only IOEx or also securityexception?

Reply:
At the moment it's only IOExceptions thrown in case the dirs cannot be created for whatever reason.